tools/mo/unit_tests/mo/back/ie_ir_ver_2/emitter_test.py

```diff
@@ -117,3 +126,143 @@ def test_serialize_old_api_map_result(self):
         self.assertTrue(serialize_res.startswith("b'<net><rt_info>"))
         self.assertTrue(serialize_res.endswith("</rt_info></net>'"))
+
+
+class TestSerialize(unittest.TestCase):
+    @staticmethod
+    def build_graph_with_gather():
+        nodes = {
+            **shaped_parameter('data', int64_array([3, 3]), {'data_type': np.float32, 'type': Parameter.op}),
+            **shaped_parameter('indices', int64_array([1, 2]), {'data_type': np.float32, 'type': Parameter.op}),
+            **valued_const_with_data('axis', int64_array(1)),
+            **regular_op_with_empty_data('gather', {'op': 'Gather', 'batch_dims': 0, 'infer': Gather.infer,
+                                                    'type': Gather.op}),
+            **result('res'),
+        }
+
+        edges = [
+            *connect('data', '0:gather'),
+            *connect('indices', '1:gather'),
+            *connect('axis', '2:gather'),
+            *connect('gather', 'res'),
+        ]
+
+        graph = build_graph(nodes, edges)
+
+        data_node = Node(graph, 'data')
+        Parameter.update_node_stat(data_node, {})
+        indices_node = Node(graph, 'indices')
+        Parameter.update_node_stat(indices_node, {})
+
+        gather_node = Node(graph, 'gather')
+        Gather.update_node_stat(gather_node, {})
+
+        res_node = Node(graph, 'res')
+        Result.update_node_stat(res_node, {})
+
+        partial_infer(graph)
+        type_infer(graph)
+
+        return graph
+
+    @staticmethod
+    def build_graph_with_maxpool():
+        graph = build_graph(
+            nodes_attrs={
+                'input': {'kind': 'op', 'op': 'Parameter', 'name': 'node', 'infer': Parameter.infer,
+                          'shape': [1, 3, 10, 10]},
+                'input_data': {'kind': 'data', 'value': None, 'shape': None},
+
+                'pool': {'kind': 'op', 'type': 'MaxPool', 'infer': Pooling.infer,
+                         'window': np.array([1, 1, 2, 2]), 'stride': np.array([1, 1, 2, 2]),
+                         'pad': np.array([[0, 0], [0, 0], [0, 0], [1, 1]]),
+                         'pad_spatial_shape': np.array([[0, 0], [1, 1]]),
+                         'pool_method': 'max', 'exclude_pad': False, 'global_pool': False,
+                         'output_spatial_shape': None, 'output_shape': None,
+                         'kernel_spatial': np.array([2, 2]), 'spatial_dims': np.array([2, 3]),
+                         'channel_dims': np.array([1]), 'batch_dims': np.array([0]),
+                         'pooling_convention': 'full', 'dilation': np.array([1, 1, 2, 2]),
+                         'auto_pad': 'valid'},
+
+                'pool_data': {'kind': 'data', 'value': None, 'shape': None},
+                'pool_data_added': {'kind': 'data', 'value': None, 'shape': None},
+                'result': {'kind': 'op', 'op': 'Result'},
+                'result_added': {'kind': 'op', 'op': 'Result', 'infer': lambda x: None}
```

Comment:
why `'infer': lambda x: None` is needed? Looks like not needed.

Reply:
agreed and removed